### PR TITLE
Comma conditionals for Job Listings (PD-1919)

### PR DESCRIPTION
**Pull request by misterjonscott**
Found a suitable set of conditions in the myblocks implementation of
the searchresult.html template and used it to add the comma logic to
the current detail page.

### CODE REVIEW

**Comment by aspidites** [comment]
I'm not pointing to solr. There a screenshot demonstrating how this changes output?

File: templates/job_detail.html [hunk 1]
```diff
@@ -32,20 +32,25 @@
...
-          <span itemprop="name">{{the_job.company}}</span>
+            <span itemprop="name">{{the_job.company}}</span>
           </span>
           <span class="direct_highlightedText">
             <span itemprop="title">{{ the_job.title|cut:"?"|cut:"~" }}</span> in 
             <span itemprop="jobLocation" itemscope itemtype="http://schema.org/Place">
               <span itemprop="address" itemscope itemtype="http://schema.org/PostalAddress">
-                {% if the_job.city != None %}
-                <span itemprop="addressLocality">{{ the_job.city }}</span>
+                {% if the_job.state_short %}
```
Comment (JLMcLaugh) [inline templates/job_detail.html]: We should keep using the_job.state. Switching to state_short is outside the scope of the ticket.

File: templates/myblocks/blocks/searchresult.html [hunk 2]
```diff
@@ -18,14 +18,14 @@ <h4 itemprop="title" {% if featured %}style="background-image:url('//d2e48ltfsb5
                             {% endif %}
                             <span class="hiringPlace" itemscope itemtype="http://schema.org/Place">
                                 <span itemprop="address" itemscope itemtype="http://schema.org/PostalAddress">
-                                    {% if job.state_short %}
+                                    {% if job.state %}
```
Comment (JLMcLaugh) [inline templates/myblocks/blocks/searchresult.html]: This shouldn't be changed either.

**Comment by aspidites** [comment]
Need a regression test. It's a template, but it should be trivial enough to render that template with various contexts and ensure the comma appears where it needs to.

File: templates/job_detail.html (hunk 1 above)
Comment (haliscooper) [inline templates/job_detail.html]: Correct me if I'm wrong, but I don't see where the state ever gets written out in this logic...

Comment (haliscooper) [inline templates/job_detail.html]: > Correct me if I'm wrong, but I don't see where the state ever gets written out in this logic...
> And a span open is left hanging from line 42?

Ignore, git cuts it off...

**Comment by JLMcLaugh** [comment]
@aspidites I agree.

**Comment by misterjonscott** [comment]
I've updated based on comments and added a test.

**Comment by aspidites** [comment]
@misterjonscott  once this is opened, you should probably resync with quality control, as you're diff is showing changes from other developers which are already in QC.

**Pull request closed without merging**